src/main/java/seedu/address/logic/commands/RedoableCommand.java

```diff
@@ -36,7 +33,7 @@ private void saveAddressBookSnapshot(Model model) {
     protected final void undo(Model model) {
         requireAllNonNull(model, this.previousAddressBook);
 
-        model.setSwappedAddressBook(this.isSwapped, this.previousAddressBook, this.previousArchiveBook);
+        model.setAddressBook(this.previousAddressBook);
 
```

Comment:
By removing saving the state of both addressbook and archivebook, undo redo is broken.